### DISCUSSION TITLE
XLOOKUP / VLOOKUP (Excel) equivalent Feature

### DISCUSSION THREAD

**Discussion by GlassedSilver** [directus/directus#8058] [Feature Requests]
I'll just quote myself from the Discord question I put forward for this endeavour of mine, because I doubt I can make someone understand a second time using much different words :P

> 
> Say I have three tables (I'm a bit stuck in classic terms here, because I don't find Directus terminology too intuitive for someone already familiar not with all, but a lot of database terminology)
> 
> Tables Figurines, Characters, Franchises (try to figure out some of my hobbies as we go on :P)
> 
> I want to assign franchises to characters and characters to figurines. These assignments would all be many-to-many. (Various franchises can be assigned to various characters and vice versa. A figurine can contain various characters (e.g. two characters related to each other as one molded item) So that's many to many as well.
> 
> Now when I create an entry for a figurine I would love to have a (read-only) field that displays the names of the franchises assigned to all characters that are assigned to this figurine. Accumulated, maybe as "tag chips".
> This should auto-fill, because the data will be present in the assigned characters' entries each.

I also explained a little example in that request, maybe it's useful to understand this idea here as well :)

> ## Table Figurines
> Normal field: Name of figurine
> M2M field: Characters Featured
> **this is the field I want to get: read "Franchises" from all Characters mentioned in the field above accumulated (ideally don't display doubles, so a cleaned up accumulated list)**
> 
> ## Table Characters
> Normal Field: Name of Character
> M2M field: Assigned Figurines
> M2M field: Franchises
> 
> ## Table Franchises
> Normal Field: Name of Franchise
> M2M field: Assigned Characters(edited)
> 
> So I want to SET the assignment in Figurines in ONE field (M2M field Characters Featured), that's where I wire together.
> 
> Then I'll have fields that display relevant data (I picked Franchises as an example here) that autofill by what I selected in the ONE M2M field I actually edit
> It's for convenience, the associated data for characters would be edited in the Characters form then
> so changes automatically appear in assigned figurines automatically as well :)

___

[Link to the original Discord question](https://discord.com/channels/725371605378924594/725817819882586132/887660893373104150) in case someone needs the reference or finds it useful :)

**Comment by GlassedSilver** [comment]
Since I'm currently working a personal spreadsheet and needed this function a day ago I remembered that the Excel terminology is probably welll known enough to use that as a lot more descriptive title for this feature idea.

The old title was way too confusing, so that's why I refer to the Excel function now. (one of my absolutely most used functions in Excel by a long shot)

**Reply by joselcvarela** [reply]
Hey @GlassedSilver 
So you have a relationship like this
A <-> B <-> C
and you want to list Cs in A view, am I correct?

If that's the case, we currently have no support for this

But we provide custom displays for developers to create their own components like what you are trying to achieve.
You can know more here: https://docs.directus.io/guides/displays/

**Reply by GlassedSilver** [reply]
That is indeed what I'm trying to achieve and to be frank given how this is basically a key feature in spreadsheets I'm baffled this isn't a default feature for a database front-end. I did find out through Discord that this is indeed not yet possible beyond what you describe, basically I just want to advocate for this to become a default feature.

As for creating my own display... I'm a bloody newbie to JS (beyond some humble script copy pasting for a few websites many years ago) and Vue and wouldn't like to self-support myself on something like this and keeping it updated in case there's a breaking change.
I'm just being honest with you. I'm well aware that's not a beloved answer, because sometimes people beyond the core team have to contribute as well, as is the root of FOSS, but I just honestly believe that someone else is probably going to do a much better job for this. It's basically the only glaring ommission I see so far in Directus and I'm really happy with everything else, lack of XLOOKUP however really holds back my commitment to my hobby database projects I'm trying to realize.

I figured I'd better draw some data models first anyhow to properly plan my schemas, haven't found a convincing FOSS solution for that yet though. Oh well, maybe I'll just doodle it all in One Note or something.

**Reply by rijkvanzanten** [reply]
As far as I can tell, assuming you have a many to many interface configured from A->B and from B->C, you can use the "related-values" display to show the items from C on A. I think this also ties into to the feature request to allow for selecting nested relational values on the tabular layout (similarly to how you already can on the cards layout) (there's a discussion about that, but I'm on my phone rn so it's a little tricky to go find)

**Reply by GlassedSilver** [reply]
Hmm, what would I put into the display template to show the value? I tried my hand at it, but I think I didn't get it. Can't seem to find the bit in the docs as well, but might be because I'm tired. :D

**Reply by rijkvanzanten** [reply]
That's a little hard to tell without knowing the full schema. The display template on the related values display for the "characters" field on "figurines" would look something like `{{ character_id.franchises.franchise_id.name }}`. The added `_id` columns here are my best guess as to what you junction tables between figurines<->characters and characters<->franchises look like

**Reply by GlassedSilver** [reply]
Yeah I was looking for a sample to adapt my own names to.

I'll look into it, thanks already! :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by GlassedSilver** [reply]
The request is 1.5 years old, has it been "bumped" to get its fair shot at this (apparently new?) requirement?

**Reply by GlassedSilver** [reply]
Really an odd decision to bulk dump ideas submitted long before that change. Especially when we're talking about functions that programmatically can extend Directus' use cases.

**Reply by rijkvanzanten** [reply]
> Really an odd decision to bulk dump ideas submitted long before that change. Especially when we're talking about functions that programmatically can extend Directus' use cases.

Happy to take any suggestions on how you would've done it differently. There were about 2,600 open feature request, of which the vast majority duplicates and requests without enough information to be able to discuss implementation details. The updated requirements make sure that feature requests A) are required to have all the information needed to be able to go over to implementing, and B) make sure that good feature requests don't get flooded and lost in the shuffle of duplicates and non-flushed out. This has actually helped out a _ton_ in making sure feature requests don't get lost, and help the community in discovering already open feature requests and upvote what's interesting to them 👍🏻
